### DISCUSSION TITLE
Add Info Box: Home Arrival Altitude and and Alternate Arrival Altitude

### DISCUSSION THREAD

**Discussion by bfcase** [XCSoar/XCSoar#1464] [Ideas]
I know every one has another info box they want added. Here are my two.
Home Arrival altitude with a sub value of  Distance.  I am looking to transition from Top Hat to XCSoar and this is the box that I currently really miss in XCSoar. Hopefully I just missed it and you can tell me I just didn't find it. So far I can only find Home Distance.

The other box I would love to have is Alternate Airport Arrival Altitude   and Distance.

Knowing I can make it to these point and have excess altitude is the most important information to me to have available.

I could probably spend quite a few hours figuring out how to make a coding contribution to XCSoar, and crudely make this myself. Where as  someone that knows what they are doing could probably do this in a very short amount of time.

Thank You

Brian Case

**Comment by kobedegeest** [comment]
You know these arrival altitudes are written on the map right? (When in reach) 

But in any case if you want those and they are in top hat already you can just copy them from there to here. If their github is nicely managed you could just pick the needed commits.
https://github.com/rdunning0823/tophat/commit/5234de3222f21a37a7d42e530fa3ae6869b677f6 
is the first one, got renamed few times after that. 

I would say give it a try yourself once you figured out how to get a commit from top hat into xcsoar you could copy over all the features you like. So it is an investment that will pay off in the long run.

**Comment by ericcarden** [comment]
Hi Brian,

Thanks for the idea. I love it! I'm a little jealous I didn't think of it myself. I, too, think the most important thing a flight computer can do is tell me whether I'm within reach of a safe place to land. You didn't overlook it, by the way, there's currently no infobox in XCSoar for home altitude difference (arrival altitude). I routinely have to touch the screen in flight while flying a task to check my arrival altitude to the home waypoint (or the best alternate). This involves panning and/or zooming, going to the alternates list, or going to the waypoints list. It would indeed be better to just see these important values in infoboxes, making it easy to check them without having to touch the screen.

I came across this "conversation" of yours about a week ago and have been digging into the code to see whether I can do this. I think I can and have accomplished part of it already (in my personal copy of the code - not yet pushed to the main project): creating a "Home altitude difference" infobox with distance shown in small print. It wasn't as simple as just copying over the Top Hat code, though. Top Hat and XCSoar have diverged a fair amount since Top Hat forked from XCSoar, and it's been 11 years since this infobox was added to Top Hat. One thing I haven't done yet is make special stuff happen when the infobox is tapped. It appears that in Top Hat, tapping this infobox brings up a small menu that lets you do stuff like "goto" the home waypoint or choose which waypoint is the home waypoint. XCSoar has no such code for home-waypoint-related infoboxes, and I haven't yet tried to bring that significant code over. How important is that to you?

My thinking is to do things in this sequence, getting each step into the main public project (at least queued for the next release) before taking the next step:

1. Add a simple "Home altitude difference" infobox (simple in that tapping it causes nothing special to happen).
2. Add a simple "Alternate 1 altitude difference" infobox.
3. (maybe) Add a simple "Alternate 2 altitude difference" infobox.
4. (maybe) Add code to make the "Home altitude difference" infobox do special stuff (like in Top Hat) when tapped.
5. (maybe) Add code to make the "Alternate" altitude difference infobox(es) do special stuff when tapped (if it's this way in Top Hat... I haven't studied that far yet).

What do you think? Thanks for the idea.

Eric

**Comment by bfcase** [comment]
Hello Eric,

Awesome and thank you for looking into this.
1 Simple Home Altitude Difference - Perfect this would be a big Improvement.
2. Alternate Altitude Difference,  - Love this idea as well
3 Alternate 2 Altitude Difference - might be a bit overkill but probably easy to add.
4. Home Altitude - Special Stuff, I actually really liked this feature of being able to change the Home airport as I moved out of range of the home airport, I would set another airport that would be an acceptable land out for me.
5. As I recall tapping the Alternate box on Top Hat, would pull up a list of reachable airports or almost reachable airports. I do find this feature very useful to quickly Identify  airports within  range that might not be on the map display.

Thanks Again.
I look forward to trying these.

Brian

Sent from my iPad

On Jun 19, 2024, at 12:13 PM, Eric Carden ***@***.***> wrote:

﻿

Hi Brian,

Thanks for the idea. I love it! I'm a little jealous I didn't think of it myself. I, too, think the most important thing a flight computer can do is tell me whether I'm within reach of a safe place to land. You didn't overlook it, by the way, there's currently no infobox in XCSoar for home altitude difference (arrival altitude). I routinely have to touch the screen in flight while flying a task to check my arrival altitude to the home waypoint (or the best alternate). This involves panning and/or zooming, going to the alternates list, or going to the waypoints list. It would indeed be better to just see these important values in infoboxes, making it easy to check them without having to touch the screen.

I came across this "conversation" of yours about a week ago and have been digging into the code to see whether I can do this. I think I can and have accomplished part of it already (in my personal copy of the code - not yet pushed to the main project): creating a "Home altitude difference" infobox with distance shown in small print. It wasn't as simple as just copying over the Top Hat code, though. Top Hat and XCSoar have diverged a fair amount since Top Hat forked from XCSoar, and it's been 11 years since this infobox was added to Top Hat. One thing I haven't done yet is make special stuff happen when the infobox is tapped. It appears that in Top Hat, tapping this infobox brings up a small menu that lets you do stuff like "goto" the home waypoint or choose which waypoint is the home waypoint. XCSoar has no such code for home-waypoint-related infoboxes, and I haven't yet tried to bring that significant code over. How important is that to you?

My thinking is to do things in this sequence, getting each step into the main public project (at least queued for the next release) before taking the next step:

  1.  Add a simple "Home altitude difference" infobox (simple in that tapping it causes nothing special to happen).
  2.  Add a simple "Alternate 1 altitude difference" infobox.
  3.  (maybe) Add a simple "Alternate 2 altitude difference" infobox.
  4.  (maybe) Add code to make the "Home altitude difference" infobox do special stuff (like in Top Hat) when tapped.
  5.  (maybe) Add code to make the "Alternate" altitude difference infobox(es) do special stuff when tapped (if it's this way in Top Hat... I haven't studied that far yet).

What do you think? Thanks for the idea.

Eric

**Comment by ericcarden** [comment]
You're welcome, Brian, and thanks for the info. Can you tell me, please, exactly what happens when you tap the home arrival altitude infobox (in Top Hat) and the step-by-step process for using it to designate a new home waypoint? As XCSoar is now, here are the two simplest procedures I know for designating a new home waypoint while flying:

Procedure 1:
1. Tap the desired new home WP on the map (after panning and/or zooming as needed).
2. In the "Map elements..." dialog that appears, tap the desired new home WP.
3. Tap the right arrow.
4. Tap "Set as New Home".

Procedure 2 (works only if the desired new home WP is currently one of the six best "alternates" as auto-selected by XCSoar):
1. Swipe a backward capital "L" (down then left) on the map.
2. Tap the desired new home WP.
3. Tap the right arrow.
4. Tap "Set as New Home".

Are these acceptable substitutes for how it's done in Top Hat starting with tapping the home arrival altitude infobox, or is the Top Hat process significantly better?

Thanks,

Eric

**Comment by bfcase** [comment]
Here are some screenshots of what happens when you tap the button and then tap the select waypoint. The gear in the top just let you change The box to a different info, box.
[image0.jpeg][image1.jpeg][image2.jpeg]
On Jun 25, 2024, at 3:48 PM, Eric Carden ***@***.***> wrote:

﻿

You're welcome, Brian, and thanks for the info. Can you tell me, please, exactly what happens when you tap the home arrival altitude infobox (in Top Hat) and the step-by-step process for using it to designate a new home waypoint? As XCSoar is now, here are the two simplest procedures I know for designating a new home waypoint while flying:

Procedure #1<https://github.com/XCSoar/XCSoar/pull/1>:

  1.  Tap the desired new home WP on the map (after panning and/or zooming as needed).
  2.  In the "Map elements..." dialog that appears, tap the desired new home WP.
  3.  Tap the right arrow.
  4.  Tap "Set as New Home".

Procedure #2<https://github.com/XCSoar/XCSoar/pull/2> (works only if the desired new home WP is currently one of the six best "alternates" as auto-selected by XCSoar):

  1.  Swipe a backward capital "L" (down then left) on the map.
  2.  Tap the desired new home WP.
  3.  Tap the right arrow.
  4.  Tap "Set as New Home".

Are these acceptable substitutes for how it's done in Top Hat starting with tapping the home arrival altitude infobox, or is the Top Hat process significantly better?

Thanks,

Eric

**Comment by bfcase** [comment]
Hello Eric,

Just sent you some screenshots of the Home Box. Pretty much just lets you select the waypoint from a list. Sorted (alphabetical, Arrive alt, or distance).
the #2 method might have the issue of, if I have a lot of close airports it might not show the airport I could reach and want to use as my new home airport.

Actually I really like the option of just selecting the airport on the map and designating at as the new Home airport. This might be the best and simplest solution.



Brian.
________________________________

**Comment by ericcarden** [comment]
Thanks. I don't see any screen shots, but maybe they aren't needed. So you consider the existing XCSoar procedure of tapping a waypoint on the map and designating it as the new home waypoint to be at least as good as the Top Hat method that starts with tapping the home arrival altitude infobox? If so, then I guess there's no need to do the work of adding the Top Hat method (for the home arrival altitude infobox) to XCSoar. Agreed?

**Comment by bfcase** [comment]
Thank you, Yes, I think the map method is just fine.

Brian

________________________________

**Comment by bfcase** [comment]
[image0.jpeg][image1.jpeg][image2.jpeg][image3.jpeg]

On Jun 26, 2024, at 6:03 PM, BRIAN CASE ***@***.***> wrote:

﻿
Thank you, Yes, I think the map method is just fine.

Brian

________________________________

**Comment by bfcase** [comment]
[IMG_3381.jpg]


[IMG_3382.jpg]


[IMG_3383.jpg]


[IMG_3384.jpg]

**Comment by ericcarden** [comment]
Thanks. I pushed this change three days ago (6/27), and it's under review. I'll report back after the review is complete.

**Comment by bfcase** [comment]
Thank you,

I have a couple weeks for flying coming up, I am looking foward to trying it.

Thanks Again.

Brian

Sent from my iPad

On Jun 30, 2024, at 3:38 PM, Eric Carden ***@***.***> wrote:

﻿

Thanks. I pushed this change three days ago (6/27), and it's under review. I'll report back after the review is complete.

**Comment by ericcarden** [comment]
This change was approved and merged into the latest code yesterday. I expect to see this new infobox appear in release version v7.44, but I have no idea when v7.44 will be released.

**Reply by lordfolken** [reply]
As soon as people help me complete https://github.com/xcsoar/xcsoar/issues?q=is%3Aopen+is%3Aissue+milestone%3Av7.44

**Comment by GuillermoHazebrouck** [comment]
This was implemented in G-NAV from the beginning... It always display the distance, altitude and required descent to the next active waypoint. Then, by touching the data panel, it splits in two, showing the home waypoint too. You can toggle between the two states just by pressing the data panel.
The arrival altitude is good to have as main source of information, but what if the GPS is not providing the altitude? This truly happens, I even had it once while flying! So in that case, you can still show the calculated required descent and use the aircraft own altimeter to estimate the arrival altitude, for which you only need to know the position.
